pkg/plugins/backendplugin/backend_plugin.go

```diff
@@ -96,3 +119,163 @@ func (p *BackendPlugin) stop() error {
 	}
 	return nil
 }
+
+// HasMetricsCollector return whether backend plugin supports collecting metrics.
+func (p *BackendPlugin) HasMetricsCollector() bool {
+	return p.diagnostics != nil
+}
+
+// CollectMetrics implements the collector.Collector interface.
+func (p *BackendPlugin) CollectMetrics(ctx context.Context, ch chan<- prometheus.Metric) error {
+	if p.diagnostics == nil {
+		return nil
+	}
+
+	if p.client == nil || p.client.Exited() {
+		return nil
+	}
+
+	res, err := p.diagnostics.CollectMetrics(ctx, &pluginv2.CollectMetrics_Request{})
+	if err != nil {
+		if st, ok := status.FromError(err); ok {
+			if st.Code() == codes.Unimplemented {
+				return nil
+			}
+		}
+
+		return err
+	}
+
+	if res == nil || res.Metrics == nil || res.Metrics.Prometheus == nil {
+		return nil
+	}
+
+	reader := bytes.NewReader(res.Metrics.Prometheus)
+	var parser expfmt.TextParser
+	families, err := parser.TextToMetricFamilies(reader)
+	if err != nil {
+		return errutil.Wrap("failed to parse collected metrics", err)
+	}
+
+	for _, mf := range families {
+		if mf.Help == nil {
+			help := fmt.Sprintf("Metric read from %s plugin", p.id)
+			mf.Help = &help
+		}
+	}
+
+	for _, mf := range families {
+		convertMetricFamily(p.id, mf, ch, p.logger)
+	}
+
+	return nil
+}
+
+// convertMetricFamily converts metric family to prometheus.Metric.
+// Copied from https://github.com/prometheus/node_exporter/blob/3ddc82c2d8d11eec53ed5faa8db969a1bb81f8bb/collector/textfile.go#L66-L165
+func convertMetricFamily(pluginID string, metricFamily *dto.MetricFamily, ch chan<- prometheus.Metric, logger log.Logger) {
```

Comment:
This may be a totally fine approach.... so ignore me if I am crazy...

Arent the exports all simple text?  Could we just export text and then search+replace a namespaced prefix?  I don't have a sense of if that is more or less fragil than trying to reconstruct the metrics in the local collector

Reply:
We need to convert metrics to a contract that prometheus golang collector expects.